src/components/ebay-carousel/examples/discrete-controlled.marko

```diff
@@ -12,12 +12,13 @@ export interface State {
 }
 
 class {
+    declare state: State;
     onCreate() {
         this.state = {
             index: 0
         }
     }
-    onMove({visibleIndexes}) {
+    onMove({visibleIndexes}: {visibleIndexes: number[]}) {
```

Comment:
If we _really_ want proper types here, we can do
```marko
import type { Input as CarouselInput } from "<ebay-carousel>";
```
and 
```suggestion
    onMove({visibleIndexes}: Parameters<NonNullable<CarouselInput["on-move"]>>[0]) {
```
or (I think)
```suggestion
    onMove: CarouselInput["on-move"] = ({visibleIndexes}) => {
```

Reply:
This can be applied all over the place in this PR, no need to do it though if you don't think it's necessary. Your method is still type safe since it will error where the function is used if the types don't match.

Reply:
Will try it out, I remember last time I tried it it didn't work.

Reply:
It works!

Reply:
That would be great, Marko 6 won't have this problem but it would be nice to have the better inference in older code. No need to complete the refactor, again your method is still type-safe